src/main/java/fi/aalto/cs/intellij/common/Resources.java

```diff
@@ -0,0 +1,71 @@
+package fi.aalto.cs.intellij.common;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Properties;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * A class that provides methods to read the resource files of the plugin application.
+ * Normally, this class should be accessed via {@code Resources.SingletonUtility.INSTANCE}.
+ */
+class Resources {
```

Comment:
I'd propose checking the [Files API](https://docs.oracle.com/javase/8/docs/api/java/nio/file/Files.html) for reading the resources, as it has many useful methods there already (and no need to test them too :wink:

Reply:
I am not sure if Files API helps a lot if we still want to access resources via class loader (instead of reading the files from the file system). `ClassLoader` provides essentially two methods to read a resource: `getResource(String)` which returns a URL and `getResourceAsStream(String)` which returns a stream. The former could be used (I'd assume) to get URL that is then opened as a file via Files API but that would just increase complexity IMO.

Not so surprisingly, it seems that `Properties` class have been designed to work in this context as there is a method `load(InputStream)`. This `Resources` class i wrote just provides a slightly simpler interface to that functionality, as it:
1. Instantiates a `Properties` object so the client code doesn't have to do that.
2. Opens and closes the stream.
3. Take care of different kind of error situations (resource doesn't exist, stream couldn't be opened, the content of the stream didn't match the properties format etc.) and indicates any failure by returning `null`. (In client code, which wants to read some properties from resources, it usually makes no difference what kind of problem there was. If those properties are not available for whatever reason, either live with it, i.e. use some defaults, or panic exit.)